### DISCUSSION TITLE
Building the application on bench start

### DISCUSSION THREAD

**Discussion by cdiethelm** [frappe/frappe-ui#133] [Q&A]
Hi, 

I've now finished creating my app in the frappe-ui development environment. 
I used this for the starter: 
https://github.com/NagariaHussain/doppio_frappeui_starter

However, I have some difficulties pushing the code to build on bench start. 

"yarn dev" works and connects to the site itself, but when I want to release the app to production, there would need to be some modifications to it, right? 

How is this solved?

**Comment by Avarsa** [comment]
https://github.com/NagariaHussain/doppio

Scroll down to "Building for production"

**Comment by j2l** [comment]
Hello, I also have this problem and tried adding package.json and `bench build --app test` but it fails:
```
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Command failed: yarn build
    at checkExecSyncError (node:child_process:885:11)
    at execSync (node:child_process:957:15)
    at run_build_command_for_apps (/home/frappe/frappe-bench/apps/frappe/esbuild/esbuild.js:500:5)
    at execute (/home/frappe/frappe-bench/apps/frappe/esbuild/esbuild.js:128:23)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 644,
  stdout: null,
  stderr: null
}
error Command failed with exit code 1.
...
subprocess.CalledProcessError: Command 'yarn run production --apps ecomm --run-build-command' returned non-zero exit status 1.
```
Would you please show me the steps and files involved?
